File: Python/convert_to_graph_lg_format.py
import csv
file_dir = "/Users/btl/Desktop/Subgraph/Datasets/"
fileName = "brachypodium-correlation-cutoff-0.98-parsed.tsv"
loc_file = file_dir + fileName

### Initlizlies variables ####
# opens the located in that directory
file = open(loc_file, "r")
temp_dict = {}
current_row_id = 0
new_file_lg = open("/Users/btl/Desktop/Subgraph/Datasets/new_file.lg","w+")
new_File = open("/Users/btl/Desktop/Subgraph/Datasets/new_file.tsv","w+")

file_reader = csv.reader(file)
# import the file row by row
for row in file_reader:
        # row is a one element list. Need to access first element
        # row[0] and split it up by a space. parts consists
        # of three elements
        parts = row[0].split("\t")
        # add entries into the Hashset
        node_source_label = parts[0]
        node_target_label = parts[1]
        weight = parts[2]
        # add the keys and values
        temp_dict.update({node_source_label:current_row_id})
        current_row_id = current_row_id + 1 # increase count by each iteration

file.close()
file = open(loc_file, "r")
file_reader = csv.reader(file)
# close the file
new_file_lg.write("v\t#\tl\n")
for row in file_reader:
        parts = row[0].split("\t")
        # add entries into the Hashset
        node_source_label = parts[0]
        node_target_label = parts[1]
        weight = parts[2]
        new_File.write(node_source_label+"\t"+node_target_label+"\t"+weight+"\n")
        # check the dictionary to get the vertex id and then
        source_id = str(temp_dict.get(node_source_label))
        target_id = str(temp_dict.get(node_target_label))
        #new_file_lg.write(source_id)
        # write the following to the file in the format
        # e source node id target node id weight
        new_file_lg.write("e\t"+(source_id)+"\t"+target_id+"\t"+weight+"\n")

# close the files
file.close()
new_file_lg.close()
new_File.close()

File: Java/convert_to_graph_lg_format.java
/**
 * Created by btl on 11/29/16.
 */

import com.apple.eawt.SystemSleepListener;
import com.sun.tracing.dtrace.ArgsAttributes;

import java.io.*;
import java.util.*;


public class convert_to_graph_lg_format{
    // Read in each row and break into different
    // components, first element, second element, third elment
    // print each one of those components to the console
    // for the user to see

    public static void main(String args[]) throws IOException {
        FileWriter fw;
        FileWriter copy_fw;
        String fName = "/Users/btl/Desktop/Subgraph/Datasets/brachypodium-correlation-cutoff-0.98-parsed.lg";
        String fiName = "/Users/btl/Desktop/Subgraph/Datasets/copy-test.tsv";
        fw = new FileWriter(fName);
        copy_fw = new FileWriter(fiName);
        fw.write("v\t#\tl" + "\n");

        String file_dir = "/Users/btl/Desktop/Subgraph/Datasets/";
        String fileName = "brachypodium-correlation-cutoff-0.98-parsed.tsv";
        String file_loc = file_dir + fileName;
        String current_row;
        int current_row_id = 0;
        int unique_vertex_id = 0;
        int vertex_id = 0;
        double weight;
        FileReader in = new FileReader(file_loc);
        BufferedReader br = new BufferedReader(in);
        // dict for vertex label and vertex id
        HashSet<String> unique_label_holder = new HashSet<String>();
        HashMap<String, Integer> unique_node_labels = new HashMap<String, Integer>();
        // write to the first line of the file
        // first line =  v # 1;
        while ((current_row = br.readLine()) != null) {
            // grab reach row and break into parts
            String[] parts = current_row.split("\\\t");
            // grab the first element in the string array parts
            String row_start_label = parts[0];
            // add to the HashSet
            unique_label_holder.add(row_start_label);
            // grab the second element in the string array parts
            String tar_node_label = parts[1];
            // get currnt size of hashset

            // add to the hashset
            unique_label_holder.add(row_start_label);
            unique_label_holder.add(tar_node_label);
        }
        // load hashmap with labels,
        // to be reference to get vertex id for drawing edges
        int temp_id = 0;

        for (String s : unique_label_holder) {
            // add to hashmap with key and value
            unique_node_labels.put(s, temp_id);
            // write the vertex info to file
            fw.write("v \t" + temp_id + "\t" + temp_id + "\n");
            temp_id++;
        }

        in = new FileReader(file_loc);
        br = new BufferedReader(in);
        while ((current_row = br.readLine()) != null) {
            String[] parts = current_row.split("\\\t");
            // grab current sorce node id
            int sr_id = unique_node_labels.get(parts[0]);
            // grab current target node id
            int tr_id = unique_node_labels.get(parts[1]);
            weight = Double.parseDouble(parts[2]);
            // write the line to file
            //System.out.println(parts[0]);
            fw.write("e\t" + sr_id + "\t" + tr_id + "\t" + weight+"\n");
            copy_fw.write(parts[0]+"\t"+parts[1] + "\t" +parts[2]+"\n");
            // grab the second element in the string array parts
        }
        fw.close();
        copy_fw.close();
    }
}





